hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSource.java

```diff
@@ -518,7 +499,7 @@ private MergeOnReadInputFormat mergeOnReadInputFormat(
         tableAvroSchema.toString(),
         AvroSchemaConverter.convertToSchema(requiredRowType).toString(),
         inputSplits,
-        conf.getString(FlinkOptions.RECORD_KEY_FIELD).split(","));
+        OptionsResolver.getRecordKeyField(conf));
```

Comment:
The change is way too complicated, can you re-illustrate the issue again? What is the use case from user and what the correct behavior is expected?

Reply:
1. The table created by the upstream write (recorded in the existing metadata) do not match the columns configured by the downstream stream read. For example, some columns do not exist, resulting in the columns not be found.
-> Verification failed, throwing exception
2. The recordkey configuration does not exist
-> Verification failed, throwing exception
3. Case problem. The columns created based on calsite in the upstream are all lowercase. If there are uppercase in the downstream, such as "eventTime", the columns will not be found.
->Uniformly converted to lowercase

Reply:
Or I split this PR？

Reply:
> Case problem. The columns created based on calsite in the upstream are all lowercase. If there are uppercase in the downstream, such as "eventTime", the columns will not be found.
->Uniformly converted to lowercase

This is not expected to be handled by Hudi, I think. At least, from the catalog layer, we should make the case-sensitivity agnostic to specific engines.

> The table created by the upstream write (recorded in the existing metadata) do not match the columns configured by the do

In `HoodieTableFactory#createDynamicTableSource`, add a sanity check for the catalog table resolved schema and the existing Hudi table schema, that should be enough I guess. Similiar with the primary key definition.

Reply:
@danny0405 cc~